### PR TITLE
Sudachen

### PR DESCRIPTION
**Pull request by sudachen**

### CODE REVIEW

File: log.go [hunk 1]
```diff
@@ -4,35 +4,48 @@ import (
...
-	return lg2.Mul(invLog2_E) // instead lg2.Div(Fixed{log2_E})
+	return b
+}
+
+func Log2(x Fixed) Fixed {
+	return Fixed{log2(x.int64)}
+}
+
+func log(x int64) int64 {
+	return mul54(log2(x), 0x2c5c85fdf473de)
```
Comment (noamnelke) [inline log.go:46]: Use a const or a comment to make it clear what `0x2c5c85fdf473de` is.

**Comment by noamnelke** [comment]
I just saw [this](https://community.spacemesh.io/t/weights-and-measures-for-blocks-and-hare-rounds/58/31?u=noam) comment in the forum and I think I get it now. However, this is very confusing for someone just reading the code - it should be explained in comments.

File: bincdf_test.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package fixed
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestFixed_BinCDF(t *testing.T) {
+	fmt.Printf("%.8f\n", BinCDF(New(34), Frac(9, 1), New(25)).Float())
```
[the remark this answers is not in the document]
Comment (sudachen) [reply bincdf_test.go:9]: It's good to know ranges for tests. Same cases in which it will be used.

File: log.go (hunk 1 above)
Comment (sudachen) [reply log.go:46]: Now It's 56 bit. I will update PR soon.

**Comment by sudachen** [comment]
All constants now defined in a 56-bit fraction format. So I don't think we need a generator for it. By the way, it's more the double-float significant bits precision.

File: lgamma.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package fixed
+
+func lgamma(x int64) int64 {
+	switch {
+	case x <= 2*oneValue:
+		switch {
+		case x <= 0:
+			panic(ErrOverflow)
+		case x == 1, x == 2:
```
Comment (noamnelke) [inline lgamma.go:9]: @sudachen I think you meant:
```golang
case x == oneValue, x == 2*oneValue:
```

**Pull request merged**